src/control/modify.js

```diff
@@ -265,15 +264,20 @@ class ModifyControl extends Control {
    * @param {*} pixel
    */
   getFeatureAtPixel(pixel) {
-    const feature = (this.map.getFeaturesAtPixel(pixel, {
-      hitTolerance: this.hitTolerance,
-      layerFilter: this.layerFilter,
-    }) || [])[0];
-
-    if (this.cursorFilter(feature)) {
-      return feature;
-    }
-    return null;
+    const features = [];
+    this.map.forEachFeatureAtPixel(
+      pixel,
+      (feature, layer) => {
+        if (this.selectFilter(feature, layer)) {
+          features.push(feature);
+        }
+      },
+      {
+        hitTolerance: this.hitTolerance,
+        layerFilter: this.layerFilter,
+      },
+    );
+    return features.length && features[0];
```

Comment:
Tip of the day (if it works;)):
```
 const feature = this.map.forEachFeatureAtPixel(
      pixel,
      (feature, layer) => {
        if (this.selectFilter(feature, layer)) {
            return feature;
         }
      },
      {
        hitTolerance: this.hitTolerance,
        layerFilter: this.layerFilter,
      },
    );
    return feature;
```

Reply:
I tested, seems to work well. But doesn't this maybe cause the wrong feature to be returned when there are multiple features at the same pixel?